source/common/upstream/cluster_manager_impl.cc

```diff
@@ -604,26 +604,22 @@ void ClusterManagerImpl::ThreadLocalClusterManagerImpl::drainConnPools(
 
 void ClusterManagerImpl::ThreadLocalClusterManagerImpl::drainConnPools(
     HostSharedPtr old_host, ConnPoolsContainer& container) {
-  for (const Http::ConnectionPool::InstancePtr& pool : container.pools_) {
-    if (pool) {
-      container.drains_remaining_++;
-    }
+  std::vector<Http::ConnectionPool::Instance*> to_be_drained;
+  for (const auto& pair : container.pools_) {
+    const Http::ConnectionPool::InstancePtr& pool = pair.second;
```

Comment:
if this code ends up staying, local var not needed, same below.

Reply:
IMO this aids in readability by making it visible what type pair.second is but I'll remove these if they stay.